src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+
+            final int totalNumberOfSNPs = snpIntervals.size();
+            final SamLocusIterator locusIterator = new SamLocusIterator(bamReader, snpIntervals,
+                    totalNumberOfSNPs < MAX_INTERVALS_FOR_INDEX);
+
+            /* set read and locus filters */
+            final List<SamRecordFilter> samFilters = Arrays.asList(new NotPrimaryAlignmentFilter(),
+                    new DuplicateReadFilter());
+            locusIterator.setSamFilters(samFilters);
+            locusIterator.setEmitUncoveredLoci(false);
+            locusIterator.setIncludeNonPfReads(false);
+            locusIterator.setMappingQualityScoreCutoff(minMappingQuality);
+            locusIterator.setQualityScoreCutoff(minBaseQuality);
+
+            logger.info("Examining " + totalNumberOfSNPs + " sites in total...");
+            int locusCount = 0;
+            for (final SamLocusIterator.LocusInfo locus : locusIterator) {
+                if (locusCount % NUMBER_OF_SITES_PER_LOGGED_STATUS_UPDATE == 0) {
+                    logger.info("Examined " + locusCount + " covered sites.");
+                }
+                locusCount++;
+
+                final int totalReadCount = locus.getRecordAndPositions().size();
+                if (totalReadCount <= readDepthThreshold) {
+                    continue;
+                }
+
+                final Map<Nucleotide, ArrayList<BaseQuality>> baseQualities = getPileupBaseQualities(locus);
+                final Nucleotide refBase = Nucleotide.valueOf(refWalker.get(locus.getSequenceIndex()).getBases()[locus.getPosition() - 1]);
+                final Nucleotide altBase = inferAltFromPileup(baseQualities, refBase);
+
+                /* calculate Het log odds */
+                final double hetLogLikelihood = getHetLogLikelihood(baseQualities, refBase, altBase);
+                final double homLogLikelihood = getHomLogLikelihood(baseQualities, refBase, altBase,
+                        DEFAULT_PRIOR_REF_HOM);
+                final double hetLogOdds = (hetLogLikelihood + FastMath.log(DEFAULT_PRIOR_HET)) -
+                        (homLogLikelihood + FastMath.log(1 - DEFAULT_PRIOR_HET));
+
+                if (hetLogOdds > hetThresholdLogOdds) {
+                    hetPulldown.add(new SimpleInterval(locus.getSequenceName(), locus.getPosition(), locus.getPosition()),
+                            baseQualities.get(refBase).size(), baseQualities.get(altBase).size(),
+                            refBase, altBase);
+                }
+            }
+
+            logger.info(locusCount + " covered sites out of " + totalNumberOfSNPs + " total sites were examined.");
+
+            return hetPulldown;
+
+        } catch (final IOException | SAMFormatException e) {
+            throw new UserException(e.getMessage());
+        }
+    }
+
+    public Pulldown getTumorHetPulldownFromNormalPulldown(final File tumorBamFile, final Pulldown normalHetPulldown) {
```

Comment:
Refactored the part for construction of `SamLocusIterator` + read and mapping quality filters. This is the only refactorable piece.